### PR TITLE
ui: update fetch-mock 

### PR DESCRIPTION
**Pull request by tamird**
This change is [Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/7857)

### CODE REVIEW

File: ui/app/containers/events.spec.tsx [hunk 1]
```diff
@@ -27,14 +27,14 @@ describe("<EventList>", function() {
   });
 
   describe("refresh", function() {
-    it("refreshes events when mounted.", function () {
+    it("refreshes events when mounted.", () => {
```
Comment (maxlang) [inline ui/app/containers/events.spec.tsx]: You didn't replace all the `describe` functions.
Why don't they need a proper closures anymore?

File: ui/app/redux/metrics.spec.ts [hunk 2]
```diff
@@ -217,13 +220,16 @@ describe("metrics reducer", function() {
       });
     });
 
-    it ("correctly responds to errors.", function () {
+    it ("correctly responds to errors.", () => {
       this.timeout(1000);
```
Comment (maxlang) [inline ui/app/redux/metrics.spec.ts]: Here especially, I'm concerned `this` won't refer to the correct thing.

File: ui/app/util/fetch-mock.ts [hunk 3]
```diff
@@ -0,0 +1,8 @@
+import "../util/object-assign";
+import * as fetchMock from "fetch-mock";
+
+fetchMock.configure({
+  sendAsJson: false,
+});
+
+export = fetchMock;
```
Comment (maxlang) [inline ui/app/util/fetch-mock.ts]: Do we want to distinguish this as a test util?

**Comment by maxlang** [comment]
Mostly :lgtm: 

We should get rid of the `Object.assign` global polyfill except for tests and make sure that changing all the tests to use arrow functions is the correct move.

---

Review status: 0 of 15 files reviewed at latest revision, 6 unresolved discussions, all commit checks successful.

---

*[ui/app/redux/ui.ts, line 31 \[r2\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/7857#-KMowLVNQabR2Wk30Ssx:-KMowLVNQabR2Wk30Ssy:-973461022) ([raw file](https://github.com/cockroachdb/cockroach/blob/fcfbf162d299d36f4a6dde59f24a11983f2d2f31/ui/app/redux/ui.ts#L31)):*
> ```XML
>     case SET_UI_VALUE:
>       let { payload } = action as PayloadAction<UISetting>;
>       return Object.assign({}, state, {[payload.key]: payload.value});
> ```

We should probably stop using `Object.assign` here like we have in the rest of the code.

New code would be:
```
state = _.clone(state);
state[payload.key] = payload.value;
return state;
```

---

*[ui/app/util/object-assign.ts, line 1 \[r2\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/7857#-KMoz2ry_Nl4PDqsZkw-:-KMoz2ry_Nl4PDqsZkw0:317619411) ([raw file](https://github.com/cockroachdb/cockroach/blob/fcfbf162d299d36f4a6dde59f24a11983f2d2f31/ui/app/util/object-assign.ts#L1)):*
> ```XML
> interface ObjectConstructor {
> ```

We should add a comment at the top explaining why this is necessary, since none of our code will use it directly if we remove it from `ui.ts`. I'm almost in favor of just adding it to the `fetch-mock` file.

---

*[ui/app/util/object-assign.ts, line 7 \[r2\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/7857#-KMoxiLlyVx3SfaHiSXX:-KMoxiLlyVx3SfaHiSXY:-1372846452) ([raw file](https://github.com/cockroachdb/cockroach/blob/fcfbf162d299d36f4a6dde59f24a11983f2d2f31/ui/app/util/object-assign.ts#L7)):*
> ```XML
> /* tslint:disable */
> 
> if (typeof Object.assign != 'function') {
> ```

Might want to mention where you got the polyfill from. I'm guessing here? https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/assign#Polyfill

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/7857#-:-KMozYvvnZFaw4Pdyzj2:1309983487)*

**Comment by tamird** [comment]
Indeed, arrow functions are a bad idea with mocha: https://github.com/mochajs/mocha/issues/1969. PTAL.

---

Review status: 0 of 15 files reviewed at latest revision, 6 unresolved discussions, all commit checks successful.

---

*[ui/app/containers/events.spec.tsx, line 30 \[r2\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/7857#-KMkiqILEMgMAmgkkT9z-r2-30:-KMp7HlPshUnQm9_saaQ:1669800184) ([raw file](https://github.com/cockroachdb/cockroach/blob/fcfbf162d299d36f4a6dde59f24a11983f2d2f31/ui/app/containers/events.spec.tsx#L30)):*
<details><summary><i>Previously, maxlang (Max Lang) wrote…</i></summary>
> You didn't replace all the `describe` functions.
> 
> Why don't they need a proper closures anymore?
> 
> 
> 
> 
</details>
You're right, backed this out.

---

*[ui/app/redux/metrics.spec.ts, line 224 \[r2\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/7857#-KMkiqILEMgMAmgkkTA2-r2-224:-KMp7JeAIFldXoEgZsPk:1451759149) ([raw file](https://github.com/cockroachdb/cockroach/blob/fcfbf162d299d36f4a6dde59f24a11983f2d2f31/ui/app/redux/metrics.spec.ts#L224)):*
<details><summary><i>Previously, maxlang (Max Lang) wrote…</i></summary>
> Here especially, I'm concerned `this` won't refer to the correct thing.
</details>
Backed this out.

---

*[ui/app/redux/ui.ts, line 31 \[r2\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/7857#-KMowLVNQabR2Wk30Ssx:-KMp7aCszFrcLLrzMFnK:-499146009) ([raw file](https://github.com/cockroachdb/cockroach/blob/fcfbf162d299d36f4a6dde59f24a11983f2d2f31/ui/app/redux/ui.ts#L31)):*
<details><summary><i>Previously, maxlang (Max Lang) wrote…</i></summary>
> We should probably stop using `Object.assign` here like we have in the rest of the code.
> 
> New code would be:
> ```
> state = _.clone(state);
> state[payload.key] = payload.value;
> return state;
> ```
</details>
Done.

---

*[ui/app/util/fetch-mock.ts, line 8 \[r2\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/7857#-KMkiqILEMgMAmgkkTA5-r2-8:-KMp7bW2sUztlMSjbSan:-1160790318) ([raw file](https://github.com/cockroachdb/cockroach/blob/fcfbf162d299d36f4a6dde59f24a11983f2d2f31/ui/app/util/fetch-mock.ts#L8)):*
<details><summary><i>Previously, maxlang (Max Lang) wrote…</i></summary>
> Do we want to distinguish this as a test util?
</details>
Meh. Do you feel that we should?

---

*[ui/app/util/object-assign.ts, line 1 \[r2\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/7857#-KMoz2ry_Nl4PDqsZkw-:-KMp7eQu67gwnKJRiuur:777579698) ([raw file](https://github.com/cockroachdb/cockroach/blob/fcfbf162d299d36f4a6dde59f24a11983f2d2f31/ui/app/util/object-assign.ts#L1)):*
<details><summary><i>Previously, maxlang (Max Lang) wrote…</i></summary>
> We should add a comment at the top explaining why this is necessary, since none of our code will use it directly if we remove it from `ui.ts`. I'm almost in favor of just adding it to the `fetch-mock` file.
</details>
Good idea, but I actually pulled this out in a .js file and a .d.ts. The TS compiler doesn't like all the implicit `any`s in this polyfill, and I don't really want to deviate from the source.

---

*[ui/app/util/object-assign.ts, line 7 \[r2\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/7857#-KMoxiLlyVx3SfaHiSXX:-KMp87Zasl0rnF0ytQDj:1176054235) ([raw file](https://github.com/cockroachdb/cockroach/blob/fcfbf162d299d36f4a6dde59f24a11983f2d2f31/ui/app/util/object-assign.ts#L7)):*
<details><summary><i>Previously, maxlang (Max Lang) wrote…</i></summary>
> Might want to mention where you got the polyfill from. I'm guessing here? https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/assign#Polyfill
</details>
Yep. Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/7857#-:-KMp8X2gjE_H91cpwCD7:675822025)*

**Comment by maxlang** [comment]
:lgtm: 

---

Review status: 0 of 16 files reviewed at latest revision, 1 unresolved discussion, all commit checks successful.

---

*[ui/app/util/fetch-mock.ts, line 8 \[r2\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/7857#-KMkiqILEMgMAmgkkTA5-r2-8:-KMyTf-VmAViWOkMLD0v:-1541959989) ([raw file](https://github.com/cockroachdb/cockroach/blob/fcfbf162d299d36f4a6dde59f24a11983f2d2f31/ui/app/util/fetch-mock.ts#L8)):*
<details><summary><i>Previously, tamird (Tamir Duberstein) wrote…</i></summary>
> Meh. Do you feel that we should?
</details>
Probably fine for now, especially given that it has mock in its name and we don't have any other testutils.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/7857#-:-KMyUDOVlbEn8fj7kzoT:980077423)*

**Comment by mrtracy** [comment]
:lgtm: 

---

Review status: 0 of 16 files reviewed at latest revision, 3 unresolved discussions, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/7857#-:-KMz0nmrtEOHSe2cbkLb:980077423)*

**Pull request merged**